### ISSUE TITLE
[Bug] Personal Sign is broken

### ISSUE THREAD

**Issue by seaona** [MetaMask/metamask-mobile#6428]
**Describe the bug**
Personal Sign is broken and I can see the following error 
`REQUIRE(_dependencyMap[6], "@metamask/controller-utils").detectSIWE) is not a function`

**Screenshots**

![image](https://github.com/MetaMask/metamask-mobile/assets/54408225/641d5cc9-9a91-4de1-92f6-fb72f8d0829d)

**To Reproduce**
_Steps to reproduce the behavior_
1. Go to test dapp
2. Connect MetaMask
3. Click Personal Sign

**Expected behavior**
It should work as expected

**Smartphone (please complete the following information):**
 - Device: Pixel 6
 - OS: Android
 - App Version : will appear in `6.7`

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

**Comment by NiranjanaBinoy** [comment]
This ticket might not be related to #6192.  This error seems to have already existed before #6192 is merged.
I tested by reverting the branch to the commit before we merged the PR, and I am still encountering this error.

The failure is due to version issues of `@metamask/controller-utils` . Currently, this package version has to be range set to `~3.0.0` with a resolution due to bigInt-related problems in mobile; [this comment](https://github.com/MetaMask/metamask-mobile/pull/6125/files#r1187475086) explains why the version range has to be limited. 

The `addUnapprovedMessage` of `PersonalMessageManager` in the core repo uses `detectSIWE` from `@metamask/controller-utils` without any differentiation between extension and mobile, which triggers this failure for us when the Personal sign is created, even though we do not have SIWE implemented in mobile side. 

Related slack thread: https://consensys.slack.com/archives/C04C0F6SSJH/p1684442093154949
